homeassistant/components/telegram_webhooks.py

```diff
@@ -0,0 +1,129 @@
+"""
+Allows utilizing telegram webhooks.
+
+See https://core.telegram.org/bots/webhooks for details
+ about webhooks.
+
+"""
+import asyncio
+import logging
+from ipaddress import ip_network
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    HTTP_BAD_REQUEST, HTTP_UNAUTHORIZED)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import CONF_API_KEY
+from homeassistant.components.http.util import get_real_ip
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['python-telegram-bot==5.3.0']
+
+EVENT_TELEGRAM_COMMAND = 'telegram.command'
+
+CONF_USER_ID = 'user_id'
+CONF_API_URL = 'api_url'
+CONF_TRUSTED_NETWORKS = 'trusted_networks'
+DEFAULT_TRUSTED_NETWORKS = [
+    ip_network('149.154.167.197/32'),
+    ip_network('149.154.167.198/31'),
+    ip_network('149.154.167.200/29'),
+    ip_network('149.154.167.208/28'),
+    ip_network('149.154.167.224/29'),
+    ip_network('149.154.167.232/31')]
+ATTR_COMMAND = 'command'
+ATTR_USER_ID = 'user_id'
+ATTR_ARGS = 'args'
+
+DEPENDENCIES = ['http']
+DOMAIN = 'telegram_webhooks'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_API_KEY, default=''): cv.string,
+        vol.Optional(CONF_API_URL, default=''): cv.string,
```

Comment:
Is not needed. Please use `"{0}/api/telegram_webhooks".format(hass.config.api.base_url)`. User Change base_url with `http` component.